src/theory/rewriter.cpp

```diff
@@ -219,6 +219,10 @@ Node Rewriter::rewriteTo(theory::TheoryId theoryId,
           TheoryId newTheory = theoryOf(newNode);
           rewriteStackTop.d_node = newNode;
           rewriteStackTop.d_theoryId = newTheory;
+          Assert(
```

Comment:
What is the motivation for this assertion?

Reply:
This is a key invariant that the rewriter should maintain: that rewriting preserves types.  Note that `isSubtypeOf` will be replaces by equality of types when subtyping is eliminated.

Reply:
Ah, OK.